pkg/devfile/resources.go

```diff
@@ -0,0 +1,237 @@
+	kubernetesComponentFilter := common.DevfileOptions{
+		ComponentOptions: common.ComponentOptions{
+			ComponentType: devfilev1.KubernetesComponentType,
+		},
+	}
+	kubernetesComponents, err := devfileObj.Data.GetComponents(kubernetesComponentFilter)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to get the kubernetes components from devfile: %v", err)
+		klog.Error(errMsg)
+		return appsv1.Deployment{}, corev1.Service{}, routev1.Route{}, err
+	}
+
+	var appendedResources parser.KubernetesResources
+	var deployment appsv1.Deployment
+	var service corev1.Service
+	var route routev1.Route
+
+	returnDeploymentOnly := true // NOTE: (TODO) once devfile samples YAML have service & route definitions, switch to false. Update tests accordingly
+
+	for _, component := range kubernetesComponents {
+		if _, ok := deployAssociatedComponents[component.Name]; ok && component.Kubernetes != nil {
+			if component.Kubernetes.Inlined == "" {
+				// If the Kubernetes YAML file does not have inline content, fall back to the dev preview way
+				// Once we have resolved the flow between frontend, backend and consume only YAML files,
+				// we can err out here instead of falling back
+
+				// Deployment
+				deploymentResource, err := GetDeployResource(name, devfileObj, common.DevfileOptions{})
+				if err != nil {
+					errMsg := fmt.Sprintf("Failed to get Deployment resource for the devfile: %v", err)
+					klog.Error(errMsg)
+					return appsv1.Deployment{}, corev1.Service{}, routev1.Route{}, err
+				}
+				appendedResources.Deployments = append(appendedResources.Deployments, deploymentResource)
+
+				dockerImagePort := devfileObj.Data.GetMetadata().Attributes.GetString("alpha.dockerimage-port", &err)
+				if err != nil {
+					errMsg := fmt.Sprintf("Failed to get the Docker image port from devfile metadata attribute 'alpha.dockerimage-port': %v", err)
+					klog.Error(errMsg)
+
+					return appsv1.Deployment{}, corev1.Service{}, routev1.Route{}, err
+				}
+
+				// Service
+				serviceResource, err := GetService(devfileObj, dockerImagePort)
+				if err != nil {
+					errMsg := fmt.Sprintf("Failed to get service for the devfile: %v", err)
+					klog.Error(errMsg)
+					return appsv1.Deployment{}, corev1.Service{}, routev1.Route{}, err
+				}
+				appendedResources.Services = append(appendedResources.Services, serviceResource)
+
+				// Route
+				routeResource := GetRouteForDockerImage(name, dockerImagePort, "/", false, nil)
+				appendedResources.Routes = append(appendedResources.Routes, routeResource)
+
+				returnDeploymentOnly = false
+
+				// break here as we are no longer interested in parsing a YAML file
+				break
+			}
+
+			src := parser.YamlSrc{
+				Data: []byte(component.Kubernetes.Inlined),
+			}
+			fs := afero.Afero{Fs: afero.NewOsFs()}
+			values, err := parser.ReadKubernetesYaml(src, fs)
```

Comment:
Good suggestion; I will put in a update to the library api method params.

but for now, I might go with `NewMemMapFs()`, dont really want to do `YAML.Unmarshal` because that would be just duplicating the devfile/library code in here to read the contents.